src/Compilers/CSharp/Portable/Binder/EmbeddedStatementBinder.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// This binder owns the scope for an embedded statement.
+    /// </summary>
+    internal sealed class EmbeddedStatementBinder : LocalScopeBinder
+    {
+        private readonly SyntaxList<StatementSyntax> _statements;
+
+        public EmbeddedStatementBinder(Binder enclosing, StatementSyntax statement)
+            : base(enclosing, enclosing.Flags)
+        {
+            _statements = new SyntaxList<StatementSyntax>(statement);
```

Comment:
To be precise, the trick used to be in LocalBinderFactory.VisitPossibleEmbeddedStatement